### ISSUE TITLE
ANCOVA output incorrect

### ISSUE THREAD

**Issue by davidrast3** [jamovi/jmv#140]
When running a 2x2x2 ANCOVA, my student noticed the 'jmv' and 'car' model outputs were different. The model has 3 IVs (3 factor variables, 2 levels each) with 2 continuous covariates and 1 continuous DV. Comparing the 'jmv' ANCOVA output to the ANCOVA output provided by 'car', SPSS, JASP, and jamovi (which all provide identical outputs), I noticed what the problem is. Or, at least I think I know what the problem is: the ANCOVA summary table variable labels are incorrectly displayed

I'm running late for a meeting so I don't have time to provide a reprex but wanted to get this on your issue tracker ASAP so it can be resolved. But here is a quick example of what we're seeing:

- The output for IV1 might be F = 2.2, p = .10 in 'car', SPSS, JASP, and jamovi, but for 'jmv' the IV1 stats are F = 8.1, p = .003. But for covariate1 in 'jmv' we see F = 2.2, p = .10, whereas in 'car', SPSS, JASP, and jamovi covariate1 F =8.1, p = .003. Thus, it appears the stats in the 'jmv' ANCOVA table are correct, but the variable labels or their order is incorrectly displayed such that covariate1 is 'jmv' should be labeled IV1 in the ANCOVA table, and IV1 should be labeled covariate1. 

Hope this makes sense. Again, sorry for any vagueness, but I'm running out of my office and wanted to get this bug submitted so I don't forget. If you need more info to reproduce this error on your end, let me know and I'll put together a proper reprex later. 

Thanks.

**Comment by jonathon-love** [comment]
hey,

thanks for reporting. could you attach some code/data which reproduces it? (or attach a .omv file?)

with thanks

**Comment by davidrast3** [comment]
Jonathon,

Here is a full reprex:


``` r
# load 'car' and 'jmv'
if(!require(car)){install.packages("car")}
#> Loading required package: car
if(!require(jmv)){install.packages("jmv")}
#> Loading required package: jmv
#> 
#> Attaching package: 'jmv'
#> The following object is masked from 'package:stats':
#> 
#>     anova

# sample data for reprex
# variable names: dv, covar1, covar2, iv1, iv2, iv3
mydf <- data.frame(
  dv = c(5, 2.66666666666667, 5.66666666666667, 5.33333333333333, 8, 7,
         5, 3.33333333333333, 7.33333333333333, 5.83333333333333,
         7.83333333333333, 7.33333333333333, 6.66666666666667, 5.5,
         6.16666666666667, 3.16666666666667, 5.16666666666667, 4.16666666666667,
         3.16666666666667, 3.16666666666667, 7.16666666666667, 4.5,
         2.66666666666667, 3.66666666666667),
  covar2 = c(19, 19, 18, 18, 20, 19, 18, 20, 20, 18, 19, 19, 17, 21, 19,
             19, 23, 20, 21, 18, 18, NA, 19, 18),
  covar1 = c(4.5, 6, 5.75, 4.5, 6.25, 5, 4, 2, 3.5, 6, 5.75, 4.25, 6.25,
             4.75, 5, 4.75, 5.5, 4.5, 5.5, 2.75, 4.5, 5.5, 1.75, 5.5),
  iv1 = as.factor(c("High", "Low", "Low", "Low", "High", "High", "High",
                    "High", "High", "High", "Low", "Low", "High",
                    "High", "High", "Low", "High", "Low", "Low", "Low", "Low",
                    "High", "High", "Low")),
  iv2 = as.factor(c("Low", "Low", "High", "Low", "High", "High", "High",
                    "Low", "Low", "Low", "Low", "Low", "Low", "Low",
                    "High", "Low", "Low", "Low", "Low", "High", "High",
                    "High", "Low", "High")),
  iv3 = as.factor(c("Low", "High", "Low", "High", "Low", "High", "High",
                    "Low", "High", "High", "Low", "High", "Low", "Low",
                    "Low", "Low", "High", "High", "Low", "High", "Low",
                    "Low", "High", "Low"))
)


## ANCOVA using 'car'
options(contrasts = c("contr.helmert", "contr.poly"))
round(Anova(lm(dv ~ covar1 + covar2 + iv1 * iv2 * iv3, data=mydf), type = 3), 3)
#> Anova Table (Type III tests)
#> 
#> Response: dv
#>             Sum Sq Df F value Pr(>F)
#> (Intercept)  2.898  1   0.802  0.387
#> covar1       5.138  1   1.422  0.254
#> covar2       0.807  1   0.223  0.644
#> iv1          7.909  1   2.189  0.163
#> iv2          0.479  1   0.132  0.722
#> iv3          0.428  1   0.118  0.736
#> iv1:iv2      1.203  1   0.333  0.574
#> iv1:iv3      0.077  1   0.021  0.886
#> iv2:iv3      1.727  1   0.478  0.502
#> iv1:iv2:iv3  0.044  1   0.012  0.914
#> Residuals   46.968 13

## ANCOVA using 'jmv' 
jmv::ancova(data = mydf,
            dep = "dv",
            factors = c("iv1", "iv2", "iv3"),
            covs = c("covar1", "covar2"))
#> 
#>  ANCOVA
#> 
#>  ANCOVA                                                                    
#>  ───────────────────────────────────────────────────────────────────────── 
#>                   Sum of Squares    df    Mean Square    F         p       
#>  ───────────────────────────────────────────────────────────────────────── 
#>    iv1                    7.9089     1         7.9089    2.1891    0.163   
#>    iv2                    0.4786     1         0.4786    0.1325    0.722   
#>    iv3                    0.4280     1         0.4280    0.1185    0.736   
#>    iv1:iv2                5.1383     1         5.1383    1.4222    0.254   
#>    iv1:iv3                0.8072     1         0.8072    0.2234    0.644   
#>    iv2:iv3                1.2032     1         1.2032    0.3330    0.574   
#>    iv1:iv2:iv3            0.0772     1         0.0772    0.0214    0.886   
#>    covar1                 1.7266     1         1.7266    0.4779    0.502   
#>    covar2                 0.0439     1         0.0439    0.0122    0.914   
#>    Residuals             46.9678    13         3.6129                      
#>  ─────────────────────────────────────────────────────────────────────────
```

**Comment by jonathon-love** [comment]
thanks! that's super-helpful.

jonathon

**Comment by jonathon-love** [comment]
Fixed in the 0.8.1.16 version of jmv, available on CRAN now.

thanks for reporting.

jonathon
